pkg/controller/operators/olm/operator.go

```diff
@@ -392,13 +407,71 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 		return nil, err
 	}
 
+
+	// setup proxy env var injection policies
+	discovery := config.operatorClient.KubernetesInterface().Discovery()
+	proxyAPIExists, err := proxy.IsAPIAvailable(discovery)
```

Comment:
For the cluster operator API, we poll for a while to see if the api is ready. We should think about doing that here too.

Reply:
yeah, in `monitor` we wait indefinitely since it'a an independently running go routine. For the operator we can retry - is 6 tries, every 10-second and then giving up a good default?

Reply:
Seems reasonable to me - we probably want to go Degraded if this fails (and we're on openshift), but I think that can be a followup in the future.